graph-node: Added StableHash impls for all  Entity update structs

That3Percent opened the pull request.

Jannis commented on graph/src/data/store/scalar.rs:
```diff
@@ -16,6 +20,51 @@ pub type BigDecimal = bigdecimal::BigDecimal;
+    // My gut feeling is that there is an API that works, and it's somewhere in
+    // the area of having the stable_hash fn yield back some *unused*
+    // SequenceNumber to the caller. Intuitively I think such an API would be
+    // significantly more error prone and will not explore further at this time.
```
What's the problem here exactly? Where are we converting between integers (or in what case would we?) and where would we want backwards-compatibility?

Jannis commented on graph/src/data/store/scalar.rs:
```diff
@@ -16,6 +20,51 @@ pub type BigDecimal = bigdecimal::BigDecimal;
+        little_endian: &int.to_bytes_le().1,
+    }
+    .stable_hash(sequence_number, state)
+}
```
Why not move this into the `stable_hash` implementation below?

And then call `BigInt::stable_hash(&int, sequence_number, state)` in the implementation for `BigDecimal`?

That3Percent replied:
Because of the orphan rules.

We can't implement `StableHash` for `bigint::BigInt`, only `data::store::scalar::BigInt`. `big_decimal_stable_hash` only has access to the former and cannot construct a `data::store::scalar::BigInt` without moving out of the `BigDecimal`.

Jannis replied:
Oh, of course. `num_bigint::BigInt` vs. our `sclar::BigInt`. Nevermind.

Jannis commented on graph/src/components/store.rs:
```diff
@@ -41,6 +42,19 @@ pub struct EntityKey {
     /// ID of the individual entity.
     pub entity_id: String,
+    // Note: Do not modify fields without making a backward compatible change to
+    // the StableHash impl (below)
```
It would be worth describing what a backward-compatible change would look like? Or is this covered in the stable_hash crate docs already?

That3Percent replied:
The RFC covers backward compatibility but should be moved into the crate docs when time allows.

That3Percent commented:
@Jannis I moved the comment, rebased, and deleted the comments that you didn't find useful. This PR is dependent on [this other PR](https://github.com/graphprotocol/stable-hash/pull/3) to fix the build so it cannot be merged yet.
